pandas/core/arrays/string_arrow.py

```diff
@@ -0,0 +1,619 @@
+
+    .. warning::
+
+       ArrowStringArray is considered experimental. The implementation and
+       parts of the API may change without warning.
+
+    Parameters
+    ----------
+    values : pyarrow.Array or pyarrow.ChunkedArray
+        The array of data.
+
+    Attributes
+    ----------
+    None
+
+    Methods
+    -------
+    None
+
+    See Also
+    --------
+    array
+        The recommended function for creating a ArrowStringArray.
+    Series.str
+        The string methods are available on Series backed by
+        a ArrowStringArray.
+
+    Notes
+    -----
+    ArrowStringArray returns a BooleanArray for comparison methods.
+
+    Examples
+    --------
+    >>> pd.array(['This is', 'some text', None, 'data.'], dtype="arrow_string")
+    <ArrowStringArray>
+    ['This is', 'some text', <NA>, 'data.']
+    Length: 4, dtype: arrow_string
+    """
+
+    _dtype = ArrowStringDtype()
+
+    def __init__(self, values):
+        self._chk_pyarrow_available()
+        if isinstance(values, pa.Array):
+            self.data = pa.chunked_array([values])
+        elif isinstance(values, pa.ChunkedArray):
+            self.data = values
+        else:
+            raise ValueError(f"Unsupported type '{type(values)}' for ArrowStringArray")
+
+        if not pa.types.is_string(self.data.type):
+            raise ValueError(
+                "ArrowStringArray requires a PyArrow (chunked) array of string type"
+            )
+
+    @classmethod
+    def _chk_pyarrow_available(cls) -> None:
+        # TODO: maybe update import_optional_dependency to allow a minimum
+        # version to be specified rather than use the global minimum
+        if pa is None or LooseVersion(pa.__version__) < "1.0.0":
+            msg = "pyarrow>=1.0.0 is required for PyArrow backed StringArray."
+            raise ImportError(msg)
+
+    @classmethod
+    def _from_sequence(cls, scalars, dtype=None, copy=False):
+        cls._chk_pyarrow_available()
+        # convert non-na-likes to str, and nan-likes to ArrowStringDtype.na_value
+        scalars = lib.ensure_string_array(
+            scalars, na_value=cls._dtype.na_value, copy=False
+        )
+        return cls(pa.array(scalars, type=pa.string(), from_pandas=True))
+
+    @property
+    def dtype(self) -> ArrowStringDtype:
+        """
+        An instance of 'ArrowStringDtype'.
+        """
+        return self._dtype
+
+    def __array__(self, dtype=None) -> np.ndarray:
+        """Correctly construct numpy arrays when passed to `np.asarray()`."""
+        return self.to_numpy(dtype=dtype)
+
+    def __arrow_array__(self, type=None):
+        """Convert myself to a pyarrow Array or ChunkedArray."""
+        return self.data
+
+    def to_numpy(
+        self, dtype=None, copy: bool = False, na_value=lib.no_default
+    ) -> np.ndarray:
+        """
+        Convert to a NumPy ndarray.
+        """
+        # TODO: copy and na_value arguments are ignored
+        return self.data.__array__(dtype=dtype)
```

Comment:
We can probably leave that for a follow-up, but I realize now that this is actually the wrong behaviour when it comes to missing values. Pyarrow will use `None`:

```
In [8]: dtype = ArrowStringDtype()

In [9]: arr = pd.array(["a", pd.NA, "b"], dtype=dtype)

In [10]: np.array(arr)
Out[10]: array(['a', None, 'b'], dtype=object)
```

While we expect it to use `pd.NA` by default.

Reply:
> We can probably leave that for a follow-up

so this relates to out discussion in https://github.com/pandas-dev/pandas/pull/35259#discussion_r513281931

to match the behaviour of StringArray, we could return pd.NA here as the default in this PR.

have now added some tests and altered the implementation of to_numpy() to pass those tests.

still more to do as the default of pd.NA will fail with some dtypes if dtype is not None. 

test_astype_roundtrip is now failing. either need to fix/finish the to_numpy() implementation here or xfail as not a base extension test before merge.

Reply:
> test_astype_roundtrip is now failing. either need to fix/finish the to_numpy() implementation here or xfail as not a base extension test before merge.

Let's xfail that for now, and leave that for a follow-up. As I am personally not fully sure the test is actually correct, since it converts the NaT to a string, not to a missing value.

Reply:
> Let's xfail that for now

ok will do that.